components/drafts/panel/panel.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useState} from 'react';
+
+import {makeIsEligibleForClick} from 'utils/utils';
+import './panel.scss';
+
+type Props = {
+    children: ({hover}: {hover: boolean}) => React.ReactNode;
+    onClick: () => void;
+};
+
+const isEligibleForClick = makeIsEligibleForClick('.file-preview.post-image__column, .hljs, code');
+
+function Panel({children, onClick}: Props) {
+    const [hover, setHover] = useState(false);
+
+    const handleMouseEnter = () => {
```

Comment:
Shouldn't we use useCallback to avoid rerenders?

Reply:
This is passed directly to an HTML element so It won't make a difference in re-rendering :)